File: SMI-SDK/Java SDK/src/pt/sapo/gis/trace/appender/SMI/SMILoggerAppenderConfigProperties.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package pt.sapo.gis.trace.appender.SMI;

import java.util.List;
import pt.sapo.gis.trace.configuration.Appender.Property;
import pt.sapo.gis.trace.configuration.AppenderConfigProperties;

/**
 *
 * @author GIS Team
 */
public class SMILoggerAppenderConfigProperties extends AppenderConfigProperties{
    
    public SMILoggerAppenderConfigProperties(List<Property> properties){         
        super(properties);
    }
    
     public String getDefaultSourceID(){
        return propertiesMap.get("defaultSourceId");
    }
}

File: SMI-SDK/.NET SDK/Library/gis/trace/appender/SMI/SMILoggerAppenderConfigProperties.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using pt.sapo.gis.trace.configuration;

namespace pt.sapo.gis.trace.appender.SMI
{
    public class SMILoggerAppenderConfigProperties : AppenderConfigProperties
    {
        public SMILoggerAppenderConfigProperties(IDictionary<String, String> properties) : base(properties) { }

        public String DefaultSourceId { get { return ContainsKey("defaultSourceId") ? this["defaultSourceId"] : null; } }
    }
}
